Plugin did not respond and Request cancelled errors in version 3.84.0 (azurerm storage account resources)

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.6.5

### AzureRM Provider Version

3.84.0

### Affected Resource(s)/Data Source(s)

azurerm_storage_account

### Terraform Configuration Files

```hcl
resource "azurerm_storage_account" "xxxxxxxxxx" {
  account_replication_type        = "ZRS"
  account_tier                    = "Standard"
  allow_nested_items_to_be_public = false
  location                        = "germanywestcentral"
  min_tls_version                 = "TLS1_2"
  name                            = "xxxxxxxx"
  resource_group_name             = "xxxxxxxx-rg"

  # If specifying network_rules, one of either ip_rules or virtual_network_subnet_ids must be specified and default_action must be set to Deny.
  network_rules {
    default_action             = "Deny"
    ip_rules                   = var.common_ip_rules
    virtual_network_subnet_ids = var.virtual_network_subnet_ids_dev
    bypass                     = ["AzureServices"]
  }

  tags = {
    ManagedBy  = "xxxxx"
    Repository = "https://xxxxxxx"
  }
}
```


### Debug Output/Panic Output

```shell

│ Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.


│ Error: Request cancelled
│
│   with azurerm_storage_account.xxxxxxxx,
│   on xxxxxx.tf line 24, in resource "azurerm_storage_account" "xxxxxxxxx":
│   24: resource "azurerm_storage_account" "xxxxxxxxxxxx" {
│
│ The plugin.(*GRPCProvider).UpgradeResourceState request was cancelled.


│ Error: Plugin did not respond
│
│   with azurerm_storage_account.xxxxxxxx,
│   on xxxxxxxx.tf line 24, in resource "azurerm_storage_account" "xxxxxxxx":
│   24: resource "azurerm_storage_account" "xxxxxxxx" {
│


│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.



Stack trace from the terraform-provider-azurerm_v3.84.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x1070b0f14]

goroutine 214 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/storage.flattenAndSetAzureRmStorageAccountPrimaryEndpoints(0x14002249080?, 0x14002480a00, 0x14001b0ab00)
	github.com/hashicorp/terraform-provider-azurerm/internal/services/storage/storage_account_resource.go:3833 +0x134
github.com/hashicorp/terraform-provider-azurerm/internal/services/storage.resourceStorageAccountRead(0x14002249080, {0x1082b7740?, 0x14001496900})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/storage/storage_account_resource.go:2229 +0x1220
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x1091c2a38?, {0x1091c2a38?, 0x1400291c510?}, 0xd?, {0x1082b7740?, 0x14001496900?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:783 +0x134
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0x14001940ee0, {0x1091c2a38, 0x1400291c510}, 0x140028a60d0, {0x1082b7740, 0x14001496900})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:1089 +0x430
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0x14000286360, {0x1091c2a38?, 0x1400291c420?}, 0x14002480e00)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/grpc_provider.go:649 +0x3e4
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0x140019d6b40, {0x1091c2a38?, 0x14003ad06f0?}, 0x1400157de60)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/tf5server/server.go:789 +0x390
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x108dc1dc0?, 0x140019d6b40}, {0x1091c2a38, 0x14003ad06f0}, 0x140014f6ee0, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:431 +0x164
google.golang.org/grpc.(*Server).processUnaryRPC(0x140002a81e0, {0x1091d43c0, 0x14001b00680}, 0x14003ac87e0, 0x14001a92690, 0x10ec6ac48, 0x0)
	google.golang.org/grpc@v1.58.3/server.go:1374 +0xb44
google.golang.org/grpc.(*Server).handleStream(0x140002a81e0, {0x1091d43c0, 0x14001b00680}, 0x14003ac87e0, 0x0)
	google.golang.org/grpc@v1.58.3/server.go:1751 +0x7c4
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.58.3/server.go:986 +0xb4
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 8
	google.golang.org/grpc@v1.58.3/server.go:997 +0x160

Error: The terraform-provider-azurerm_v3.84.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Expected Behaviour

- Plan should work in version 3.84.0
- With previous azurerm TF provider works as a charm. This is only happening with 3.84.0. Tested with 3.83.0 and 3.82.0.


### Actual Behaviour

panic. ALL azurerm_storage_account resources are affected.

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/storage/storage_account_resource.go
+++ b/internal/services/storage/storage_account_resource.go
@@ -3831,7 +3831,7 @@ func flattenAndSetAzureRmStorageAccountPrimaryEndpoints(d *pluginsdk.ResourceDat
 
 	// below null check is to avoid nullpointer scenarios when either of publish_internet_endpoints
 	// or publish_microsoft_endpoints or both aren't set
-	if routingInputs != nil && *routingInputs.PublishInternetEndpoints {
+	if routingInputs != nil && routingInputs.PublishInternetEndpoints != nil && *routingInputs.PublishInternetEndpoints {
 		if err := setEndpointAndHost(d, "primary", primary.InternetEndpoints.Blob, "blob_internet"); err != nil {
 			return err
 		}
@@ -3849,7 +3849,7 @@ func flattenAndSetAzureRmStorageAccountPrimaryEndpoints(d *pluginsdk.ResourceDat
 		}
 	}
 
-	if routingInputs != nil && *routingInputs.PublishMicrosoftEndpoints {
+	if routingInputs != nil && routingInputs.PublishMicrosoftEndpoints != nil && *routingInputs.PublishMicrosoftEndpoints {
 		if err := setEndpointAndHost(d, "primary", primary.MicrosoftEndpoints.Blob, "blob_microsoft"); err != nil {
 			return err
 		}

